File: backend/models/categoriaCaminhadaCorridaModel.js
const { DataTypes } = require('sequelize');
const db = require('../database/db');

const CategoriaCaminhadaCorrida = db.define('CategoriaCaminhadaCorrida', {
    ID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    DISTANCIA: {
        type: DataTypes.STRING,
        allowNull: false,
        primaryKey: true
    },
    CHAVE: {
        type: DataTypes.STRING,
        allowNull: false
    },
    DESCRICAO: {
        type: DataTypes.STRING,
        allowNull: false
    },
    SITUACAO: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    CRIADO_EM: {
        type: DataTypes.DATE,
        allowNull: false,
        defaultValue: DataTypes.NOW
    },
    ATUALIZADO_EM: {
        type: DataTypes.DATE,
        allowNull: true,
        defaultValue: DataTypes.NOW
    }
}, {
    tableName: 'CATEGORIA_CAMINHADA_CORRIDA',
    timestamps: true,
    createdAt: 'CRIADO_EM',
    updatedAt: 'ATUALIZADO_EM'
});

module.exports = CategoriaCaminhadaCorrida;

File: frontend/lib/models/categoria_caminhada_corrida_model.dart
class CategoriaCaminhadaCorrida {
  final int id;
  final String distancia;
  final String chave;
  final String descricao;
  final bool situacao;

  CategoriaCaminhadaCorrida({
    required this.id,
    required this.distancia,
    required this.chave,
    required this.descricao,
    required this.situacao,
  });

  factory CategoriaCaminhadaCorrida.fromJson(Map<String, dynamic> json) {
    return CategoriaCaminhadaCorrida(
      id: json['ID'],
      distancia: json['DISTANCIA'],
      chave: json['CHAVE'],
      descricao: json['DESCRICAO'],
      situacao: json['SITUACAO'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'ID': id,
      'DISTANCIA': distancia,
      'CHAVE': chave,
      'DESCRICAO': descricao,
      'SITUACAO': situacao,
    };
  }
}
